### ISSUE TITLE
Tensor slicing with boolean numpy mask wrong

### ISSUE THREAD

**Issue by Stannislav** [pytorch/pytorch#25176]
## 🐛 Bug
Numpy arrays of dtype `np.bool` should be interpreted as masks when slicing torch arrays, just like tensors of dtype `torch.bool` are. Insead, such arrays are interpreted as indices.

## To Reproduce

```python
t = torch.tensor(range(5))
mask = np.array([False, True, True, False, True])

print(t[mask.astype(np.bool)])
print(t[mask.astype(np.uint8)])
```
Output
```bash
tensor([0, 1, 1, 0, 1])
tensor([1, 2, 4])
```
## Expected behavior
The expected behaviour is that the result is the same as for slicing with torch tensors with the same dtype, where for the input
```python
t = torch.tensor(range(5))
mask = torch.tensor([False, True, True, False, True])

print(t[mask.to(torch.bool)])
print(t[mask.to(torch.uint8)])
```
we get the expected output
```bash
tensor([1, 2, 4])
tensor([1, 2, 4])
```

## Environment
 - PyTorch Version: 1.2.0a0+0885dd2
 - OS: Ubuntu 16.04.6 LTS
 - How you installed PyTorch: Nvidia docker image (`FROM nvcr.io/nvidia/pytorch:19.06-py3`)
 - Python version: 3.6
 - CUDA/cuDNN version: Nvidia driver version: 384.183, ` /usr/lib/x86_64-linux-gnu/libcudnn.so.7.6.0`
 - GPU models and configuration: Tesla V100

**Comment by izdeby** [comment]
Already fixed. 
Repro code: 
```
import torch
import numpy as np
t = torch.tensor(range(5))
mask = np.array([False, True, True, False, True])

print(t[mask.astype(np.bool)])
print(t[mask.astype(np.uint8)])
```

output: 
_tensor([1, 2, 4])
../aten/src/ATen/native/IndexingUtils.h:20: UserWarning: indexing with dtype torch.uint8 is now deprecated, please use a dtype torch.bool instead.
tensor([1, 2, 4])_
